content/en/tools-corda5/contract-testing/_index.md

```diff
@@ -0,0 +1,104 @@
+---
+date: '2023-08-1237'
+title: "Contract Testing"
+project: corda5-tools
+section_menu: corda5-tools
+version: tools
+menu:
+  corda5-tools:
+    weight: 1000
+---
+
+The Contract Testing library enables {{< tooltip >}}CorDapp{{< /tooltip >}} Developers to test {{< tooltip >}}UTXO{{< /tooltip >}} contracts locally, at an early stage of the development cycle. With this library, you can check that your CorDapp contracts behave as expected before you write the flows.
+
+## Prerequisites
+
+* {{< tooltip >}}CSDE{{< /tooltip >}} — contains Contract Testing library [examples](#examples).
+* Contract and state classes
+
+## Importing the Library
+
+To import the library to your CorDapp, add it to your `contracts` `build.gradle` dependencies:
+```
+testImplementation "com.r3.corda.ledger.utxo:contract-testing:0.9.0”
+```
+
+If you are using Kotlin, you can also add the Kotlin DSL:
+```
+testImplementation "com.r3.corda.ledger.utxo:contract-testing-kotlin:0.9.0”
+```
+
+You can now extend the `ContractTest` interface.
+
+## Testing with the Library
+
+All tests should extend `ContractTest` as a starting point. This provides the `ledgerService` and `signingService` and also some useful
+test data:
+
+<style>
+table th:first-of-type {
+    width: 40%;
+}
+table th:nth-of-type(2) {
+    width: 60%;
+}
+</style>
+
+| X.500 Legal Identities | Public Keys |
+|-----------------------|-------------|
+| aliceName             | aliceKey    |
+| bobName               | bobKey      |
+| charlieName           | charlieKey  |
+| daveName              | daveKey     |
+| eveName               | eveKey      |
+| bankAName             | bankAKey    |
+| bankBName             | bankBKey    |
+| notaryName            | notaryKey   |
+
+### Happy Path Testing
+
+If you expect your test to pass the contract validation, use `assertVerifies`. For example:
+```java
```

Comment:
please indent code snippets w/ 4 spaces (or 8)

Reply:
> please indent code snippets w/ 4 spaces (or 8)

Do you mean the whole code block? Or just particular lines?

Reply:
each code block should be relatively indented by 4 spaces instead of 2

Reply:
Have updated EACH line (except the first) of both code blocks.